Library/Homebrew/formulary.rb

```diff
@@ -191,6 +193,27 @@ def self.from_rack(rack, spec=:stable)
   end
 
   def self.canonical_name(ref)
+    case ref
+    when %r{^([\w+-.]+)$}
+      return ref.downcase if (HOMEBREW_CELLAR/ref.downcase).directory?
+      if (link = (HOMEBREW_CELLAR/ref.downcase).resolved_path).directory?
+        basename = link.basename
+        if FORMULA_RENAMES[ref.downcase] == basename
+          return basename
+        end
+
+        if Tap.detect { |tap| tap.formula_renames[ref.downcase] == basename }
+          return basename
+        end
+      end
+    when HOMEBREW_TAP_FORMULA_REGEX
+      user, repo, name = ref.split("/", 3).map(&:downcase)
+      tap = Tap.new user, repo.sub(/^homebrew-/, "")
+      if (link = (HOMEBREW_CELLAR/name).resolved_path).directory?
+        return basename if tap.formula_renames[name] == link.basename
+      end
+    end
+
```

Comment:
I don't think you should change this method. `canonical_name` has completely different semiotical meaning than `Formulary#from_rack` or `ARGV#resolved_formulae`. That is it should return the same name regardless current installation.